From f3bc1bf4fac7a1fd159b3fb4bebc04d9a7684d9d Mon Sep 17 00:00:00 2001
From: zuoqian <zuoqian2032@phytium.com.cn>
Date: Tue, 23 Jan 2024 09:24:03 +0800
Subject: [PATCH 087/136] BACKPORT: PHYTIUM: drivers: fix build errors

1. spi: Rename SPI_MASTER_GPIO_SS to SPI_CONTROLLER_GPIO_SS 82238
2. net: stmmac: clarify difference between "interface" and "phy_interface"  a014c3555
3. net: stmmac: convert plat->phylink_node to fwnode   e80af2a
4. net: stmmac: Make stmmac_pltfr_remove() return void 3246627f
5. Remove IOMMU_CAP_INTR_REMAP, instead rely on the interrupt subsystem 143c7bc
6. i2c: Switch .probe() to not take an id parameter 03c835f498b54
7. drivers: gpio: add #include <linux/seq_file.h>

Signed-off-by: zuoqian <zuoqian2032@phytium.com.cn>

Link: https://gitee.com/phytium_embedded/phytium-linux-kernel/commit/52a27ae1196d3714b22e430fb43ab9ac2e55e914
[Kexy: Dropped nonexistent files]
Signed-off-by: Kexy Biscuit <kexybiscuit@aosc.io>
---
 .../net/ethernet/stmicro/stmmac/dwmac-phytium.c  | 16 +++++++---------
 1 file changed, 7 insertions(+), 9 deletions(-)

diff --git a/drivers/net/ethernet/stmicro/stmmac/dwmac-phytium.c b/drivers/net/ethernet/stmicro/stmmac/dwmac-phytium.c
index db1672deae82..86f491ea569e 100644
--- a/drivers/net/ethernet/stmicro/stmmac/dwmac-phytium.c
+++ b/drivers/net/ethernet/stmicro/stmmac/dwmac-phytium.c
@@ -73,14 +73,14 @@ static int phytium_dwmac_probe(struct platform_device *pdev)
 	if (plat->phy_interface < 0)
 		return plat->phy_interface;
 
-	plat->interface = phytium_get_mac_mode(fwnode);
-	if (plat->interface < 0)
-		plat->interface = plat->phy_interface;
+	plat->mac_interface = phytium_get_mac_mode(fwnode);
+	if (plat->mac_interface < 0)
+		plat->mac_interface = plat->phy_interface;
 
 	/* Configure PHY if using device-tree */
 	if (pdev->dev.of_node) {
 		plat->phy_node = of_parse_phandle(np, "phy-handle", 0);
-		plat->phylink_node = np;
+		plat->port_node = of_fwnode_handle(np);
 	}
 
 	if (pdev->dev.of_node) {
@@ -185,16 +185,14 @@ static int phytium_dwmac_probe(struct platform_device *pdev)
 	return stmmac_dvr_probe(&pdev->dev, plat, &stmmac_res);
 }
 
-int phytium_dwmac_remove(struct platform_device *pdev)
+void phytium_dwmac_remove(struct platform_device *pdev)
 {
-	int ret;
 	struct net_device *ndev = platform_get_drvdata(pdev);
 	struct stmmac_priv *priv = netdev_priv(ndev);
 	struct plat_stmmacenet_data *plat = priv->plat;
 
-	ret = stmmac_pltfr_remove(pdev);
+	stmmac_pltfr_remove(pdev);
 	clk_unregister_fixed_rate(plat->stmmac_clk);
-	return ret;
 }
 
 #ifdef CONFIG_OF
@@ -215,7 +213,7 @@ MODULE_DEVICE_TABLE(acpi, phytium_dwmac_acpi_ids);
 
 static struct platform_driver phytium_dwmac_driver = {
 	.probe = phytium_dwmac_probe,
-	.remove = phytium_dwmac_remove,
+	.remove_new = phytium_dwmac_remove,
 	.driver = {
 		.name		= "phytium-dwmac",
 		.of_match_table	= of_match_ptr(phytium_dwmac_of_match),
-- 
2.47.0

